### PR TITLE
Api create post

### PR DESCRIPTION
**Pull request by ghost**
## Description
Api Create Post
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
 NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- explain insert into `posts` (`type`, `content`, `rate`, `user_id`, `product_id`, `status`, `updated_at`, `created_at`) values (2, "nakqi", null, 2, 3, 0," 2018-06-22 07:56:20","2018-06-22 07:56:20")\G
*************************** 1. row ***************************
           id: 1
  select_type: INSERT
        table: posts
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: NULL
     filtered: NULL
        Extra: NULL
1 row in set (0.00 sec)

### CODE REVIEW

File: app/Http/Requests/Api/CreatePostRequest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'type' => 'in:' . Post::COMMENT . ',' . Post::REVIEW,
+            'rate'           => 'numeric|min:1|max:5',
+            'content'          => 'required|string',
```
Comment (at-thaitran) [inline app/Http/Requests/Api/CreatePostRequest.php]: nên dùng `interger` vì value mình truyền vào là số nguyên

File: app/Http/Controllers/Api/PostController.php [hunk 2]
```diff
@@ -29,4 +31,27 @@ public function destroy(Post $post)
...
+        $user = Auth::user();
+        $input = $request->only('type', 'content');
+        if ($input['type'] == Post::REVIEW) {
+            $input['rate'] = $request->rating;
+        }
+        $input['user_id'] = $user->id;
+        $input['product_id'] = $product->id;
+        $input['status'] = Post::DISABLE;
+        $post = Post::create($input);
+        $post = $post->load('user.userInfo');
```
Comment (at-thaitran) [inline app/Http/Controllers/Api/PostController.php]: thiếu tính trung bình rate và sum rate

File: app/Http/Requests/Api/CreatePostRequest.php (hunk 1 above)
Comment (at-sonle) [reply app/Http/Requests/Api/CreatePostRequest.php]: ok

File: app/Http/Controllers/Api/PostController.php (hunk 2 above)
Comment (at-sonle) [reply app/Http/Controllers/Api/PostController.php]: cái này tính ở admin lúc duyệt post

**Comment by at-thaitran** [comment]
LGTM

**Comment by at-vietpham** [comment]
LGTM 🕋

**Pull request merged**